### PR TITLE
e2e: wait for upgraded pods to be running during upgrade-tests

### PR DESCRIPTION
**Pull request by Rakshith-R**
This commit calls `waitForDaemonSets` and `waitForDeploymentComplete`
after upgrading to wait for csi driver pods to be in running state
for both rbd and cephfs upgrade tests.

Signed-off-by: Rakshith R <rar@redhat.com>

~**Changes in `rook.sh`,`build.env` and `minikube.sh` will be removed**~

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

File: e2e/upgrade-cephfs.go
Comment (Madhu-1) [inline e2e/upgrade-cephfs.go:237]: check is present here https://github.com/ceph/ceph-csi/blob/devel/e2e/upgrade-cephfs.go#L147-L158?

Comment (Rakshith-R) [reply e2e/upgrade-cephfs.go:237]: > check is present here https://github.com/ceph/ceph-csi/blob/devel/e2e/upgrade-cephfs.go#L147-L158?

This check is for older cephcsi version in upgrade tests. 

The wait I've added is for after deletion and cephcsi image is upgraded to the present changes.

**Review by Madhu-1 — COMMENTED** [review]
E2E changes LGTM. will approve once other changes are reverted

**Review by Madhu-1 — DISMISSED** [review]

**Comment by Madhu-1** [comment]
Lets wait for #1998 to merge  before restarting the E2E

**Comment by humblec** [comment]
lgtm. Thanks

**Comment by Rakshith-R** [comment]
rebased it on top of devel. PTAL @Madhu-1 @humblec

**Review by Madhu-1 — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Pull request merged**
